File: Php/app/Cashgame/Running/StatusItemModel.php
namespace app\Cashgame\Running{
	use core\Timer;
	use entities\Homegame;
	use app\Urls\CashgameReportUrlModel;
	use app\Urls\CashgameCashoutUrlModel;
	use app\Urls\CashgameBuyinUrlModel;
	use core\Util;
	use app\Urls\CashgameActionUrlModel;
	use core\Globalization;
	use entities\CashgameResult;
	use entities\Cashgame;

	class StatusItemModel{

		private $result;
		public $name;
		public $playerUrl;
		public $buyin;
		public $stack;
		public $winnings;
		public $time;
		public $winningsClass;
		public $managerButtonsEnabled;
		public $buyinUrl;
		public $reportUrl;
		public $cashoutUrl;
		public $hasCashedOut;

		public function __construct(Homegame $homegame,
									Cashgame $cashgame,
									CashgameResult $result,
									$isManager,
									Timer $timer){
			result = $result;
			$player = $result.getPlayer();
			if($player != null){
				name = $player.getDisplayName();
				playerUrl = new CashgameActionUrlModel($homegame, $cashgame, $player);
				if($isManager){
					buyinUrl = new CashgameBuyinUrlModel($homegame, $player);
					reportUrl = new CashgameReportUrlModel($homegame, $player);
					cashoutUrl = new CashgameCashoutUrlModel($homegame, $player);
				}
			}
			buyin = Globalization::formatCurrency($homegame.getCurrency(), $result.getBuyin());
			stack = Globalization::formatCurrency($homegame.getCurrency(), $result.getStack());
			$winnings = $result.getWinnings();
			winnings = Globalization::formatResult($homegame.getCurrency(), $winnings);
			$lastReportedTime = $result.getLastReportTime();
			if($lastReportedTime != null){
				$secondsAgo = $timer.getTime().getTimestamp() - $lastReportedTime.getTimestamp();
				time = Globalization::formatTimespan($secondsAgo);
			}
			winningsClass = Util::getWinningsCssClass($winnings);
			hasCashedOut = $result.getCashoutTime() != null;
			managerButtonsEnabled = $isManager;
		}

	}

}

File: Web/Models/CashgameModels/Running/StatusItemModel.cs
using Core.Classes;
using Infrastructure.System;
using Web.Models.UrlModels;

namespace Web.Models.CashgameModels.Running{

	public class StatusItemModel{

	    public string Name { get; set; }
	    public UrlModel PlayerUrl { get; set; }
		public string Buyin { get; set; }
		public string Stack { get; set; }
		public string Winnings { get; set; }
		public string Time { get; set; }
		public string WinningsClass { get; set; }
		public bool ManagerButtonsEnabled { get; set; }
		public UrlModel BuyinUrl { get; set; }
		public UrlModel ReportUrl { get; set; }
		public UrlModel CashoutUrl { get; set; }
		public bool HasCashedOut { get; set; }

		public StatusItemModel(Homegame homegame, Cashgame cashgame, CashgameResult result, bool isManager, ITimeProvider timeProvider){
			var player = result.Player;
			if(player != null){
				Name = player.DisplayName;
				PlayerUrl = new CashgameActionUrlModel(homegame, cashgame, player);
				if(isManager){
					BuyinUrl = new CashgameBuyinUrlModel(homegame, player);
					ReportUrl = new CashgameReportUrlModel(homegame, player);
					CashoutUrl = new CashgameCashoutUrlModel(homegame, player);
				}
			}
			Buyin = Globalization.FormatCurrency(homegame.Currency, result.Buyin);
			Stack = Globalization.FormatCurrency(homegame.Currency, result.Stack);
			Winnings = Globalization.FormatResult(homegame.Currency, result.Winnings);
			var lastReportedTime = result.LastReportTime;
			if(lastReportedTime.HasValue){
				var timespan = timeProvider.GetTime() - lastReportedTime.Value;
				Time = Globalization.FormatTimespan(timespan);
			}
			WinningsClass = Util.GetWinningsCssClass(result.Winnings);
			HasCashedOut = result.CashoutTime != null;
			ManagerButtonsEnabled = isManager;
		}

	}

}